app/mobile/lib/view/resource/edit_resource_page.dart

```diff
@@ -0,0 +1,154 @@
+                    child: TextFormField(
+                      controller: viewModel.controller_title,
+                      decoration: const InputDecoration(
+                        enabledBorder: const OutlineInputBorder(
+                            borderSide:
+                                const BorderSide(color: CustomColors.main)),
+                        border: OutlineInputBorder(),
+                        labelText: 'Resource Name',
+                      ),
+                      validator: (value) {
+                        if (value == null || value.isEmpty) {
+                          return 'Please Enter Some Text';
+                        }
+                        return null;
+                      },
+                    ),
+                  ),
+                  Padding(
+                    padding:
+                        const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
+                    child: Align(
+                      alignment: Alignment.centerLeft,
+                      child: Container(
+                          child: Text(
+                        "Resource Body:",
+                        style: TextStyle(
+                          fontSize: 16,
+                        ),
+                      )),
+                    ),
+                  ),
+                  viewModel.isLoading
+                      ? loadingIndicator()
+                      : Padding(
+                          padding: const EdgeInsets.symmetric(horizontal: 8),
+                          child: Container(
+                            decoration: BoxDecoration(
+                              color: const Color.fromARGB(255, 242, 241, 248),
+                              borderRadius:
+                                  BorderRadiusDirectional.circular(10),
+                              border: Border.all(color: CustomColors.main),
+                            ),
+                            child: markdownInput(null, viewModel.controller,
+                                loading: viewModel.isLoading,
+                                maxLines: 20,
+                                sendText: '',
+                                sendIcon: Icons.abc,
+                                withBorderRadius: true),
+                          ),
+                        ),
+                  GestureDetector(
+                    onTap: () async {
+                      Resource? created = await contentService
+                          .editResource(viewModel.controller_title.text, viewModel.controller.text, resourceId: r.id);
+                      if (created == null) {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(
+                              content: Text('Could Not Edit Resource!')),
+                        );
+                        viewModel.notifyListeners();
+                      }
+                      if (created!.id.isNotEmpty) {
```

Comment:
This if statement is not correct. When the created = null, checking content as nonnullable creates an error.  I think changing it with following would be enough for solving it. 

```
if (created != null) {
    if (created!.id.isNotEmpty) {
        ScaffoldMessenger.of(context).showSnackBar(
            const SnackBar(content: Text('Successfully Editted!')),);
        Navigator.pop(context);
        viewModel.notifyListeners();
   }
} else {
    ScaffoldMessenger.of(context).showSnackBar(
         const SnackBar( content: Text('Could Not Edit Resource!')),);
         viewModel.notifyListeners();
 }
 ```

Reply:
I committed the changes in here: [843d3db](https://github.com/bounswe/bounswe2022group3/commit/843d3db978706dec3e9e230e10a7feeafcb97dba)

Reply:
Thank you for fixing my error :) I guess I was very distracted while writing this.